### PR TITLE
Switch to Azure SDK v2

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
Migrate the cleanup logic to use Azure SDK v2 since v1 is deprecated

Followed instructions in https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md

co-authored by @marosset

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/assign @aramase 

We will add support for Managed Identity in a separate PR

**Comment by CecileRobertMichon** [comment]
/hold

Please hold merge until we validate this

File: main.go [hunk 1]
```diff
@@ -95,37 +96,36 @@ func main() {
...
-		return fmt.Errorf("Error when listing all resource groups: %v", err)
-	}
-	for list := listComplete; list.NotDone(); err = list.Next() {
+
+	pager := r.NewListPager(nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
 		if err != nil {
-			return fmt.Errorf("Error when iterating resource groups: %v", err)
+			log.Fatalf("failed to advance page: %v", err)
```
Comment (aramase) [inline main.go:106]:
```suggestion
			return fmt.Errorf("failed to advance page: %w", err)
```

Return an error here so `fatal` is only in the caller?

File: main.go [hunk 2]
```diff
@@ -152,18 +152,17 @@ func shouldDeleteResourceGroup(rg resources.Group, ttl time.Duration) (string, b
...
-		return nil, err
+func getResourceGroupClient(clientID, clientSecret, tenantID, subscriptionID string) (*armresources.ResourceGroupsClient, error) {
+	options := arm.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: cloud.AzurePublic,
+		},
 	}
 
-	armSpt, err := adal.NewServicePrincipalToken(*oauthConfig, clientID, clientSecret, env.ServiceManagementEndpoint)
+	credential, err := azidentity.NewClientSecretCredential(tenantID, clientID, clientSecret, nil)
```
Comment (aramase) [inline main.go:162]: Need to check for the error here

**Review by aramase — COMMENTED** [review]
`main_test.go` also needs to be updated!

File: Dockerfile [hunk 3]
```diff
@@ -1,3 +1,3 @@
-FROM alpine:3.10.3
+FROM alpine:3.18
```
[the remark this answers is not in the document]
Comment (CecileRobertMichon) [reply Dockerfile:1]: tried with distroless but was running into issues in testing, will leave as alpine for now and try to switch it in a separate PR

**Comment by CecileRobertMichon** [comment]
@aramase addresses all comments, test is passing, this is ready

File: main.go [hunk 4]
```diff
@@ -95,37 +96,36 @@ func main() {
...
-				continue
-			}
-
-			_, err = r.Delete(ctx, rgName)
-			if err != nil {
-				log.Printf("Error when deleting %s: %v", rgName, err)
+		for _, rg := range nextResult.Value {
+			rgName := *rg.Name
+			if age, ok := shouldDeleteResourceGroup(rg, ttl); ok {
+				// Does not wait when deleting a resource group
```
Comment (jackfrancis) [inline main.go:111]: Not sure what this comment means

Comment (marosset) [reply main.go:111]: i think it means we're just doing a fire-and-forget operation. We don't wait for the deletions to finish or check the status

Comment (CecileRobertMichon) [reply main.go]: clarified comment

**Review by aramase — APPROVED** [review]
LGTM

**Pull request merged**